File: LeetCode75/TwoPointers/392_is_subsequence.cs
public class Solution {
    public bool IsSubsequence(string s, string t) {
        if (s.Length == 0) return true;
        var subSequence = s.ToList();
        for(int i = 0; i < t.Length; i++){
            if(t[i] == subSequence[0]){
                subSequence.RemoveAt(0);
                if (subSequence.Count == 0) return true;
            }
        }
        return subSequence.Count == 0;
    }
}

File: LeetCode75/TwoPointers/392_is_subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        std::string::iterator s_it = s.begin();
        for(char t_char : t){
            if (s_it == s.end()) {
                return true;
            }

            if(t_char == *s_it){
                s_it++;
            }
        }
        return s_it == s.end();
    }
};